### PR TITLE
Added navigation and fixes

### PR DESCRIPTION
**Pull request by martinailieva**
Added main navigation and fixes on other pages

### CODE REVIEW

**Comment by tsanim** [comment]
Please commit new changes which are not "that" WIP. :D

**Review by dianakirova — COMMENTED** [review]
Please, check my comments :)

File: src/components/AuthForm/index.tsx [hunk 1]
```diff
@@ -5,6 +5,8 @@ import { Button, FloatingLabel, Form } from "react-bootstrap";
 import AuthContext from "store/auth-context";
 import styled from "styled-components";
 import AllRoutes from "pages/AllRoutes";
+import { getAuth, signInWithEmailAndPassword } from "firebase/auth";
```
Comment (tsanim) [inline src/components/AuthForm/index.tsx:8]: Use relative path for imports!

File: src/components/AuthForm/index.tsx [hunk 2]
```diff
@@ -25,15 +27,31 @@ const AuthForm: React.FC = () => {
...
   const submitHandler = (e: FormEvent) => {
     e.preventDefault();
-    const enteredEmail = emailInputRef?.current?.value;
-    const enteredPassword = passwordInputRef?.current?.value;
-
+    const enteredEmail = emailInputRef?.current?.value || "";
+    const enteredPassword = passwordInputRef?.current?.value || "";
+    // test(enteredEmail, enteredPassword);
     setIsLoading(true);
     let url;
```
Comment (tsanim) [inline src/components/AuthForm/index.tsx:56]: Move all login, register logic in a service file. And here in the component just call that function.

**Comment by tsanim** [comment]
Do you really need `dummyData.ts` ?

**Comment by tsanim** [comment]
`useEffect()` should be positioned just before return statements. Please fix all these occasions.

File: src/store/auth-context.tsx [hunk 3]
```diff
@@ -28,6 +33,7 @@ const AuthContext = React.createContext<AuthContextType>({
   logout: () => {
     null;
   },
+  userInfo: { email: "", displayName: "", id: "" },
```
Comment (tsanim) [inline src/store/auth-context.tsx:36]:
```
 login: () => {},
 logout: () => {},
  ```
Is enough

File: src/store/auth-context.tsx [hunk 4]
```diff
@@ -46,6 +52,7 @@ const retrieveStoredToken = () => {
   const remainingTime = calculateRemainingTime(storedExpirationDate);
```
Comment (tsanim) [inline src/store/auth-context.tsx:52]: type `retrieveStoredToken`

File: src/store/auth-context.tsx [hunk 5]
```diff
@@ -64,22 +71,27 @@ export const AuthContextProvider: React.FC<PropsWithChildren<unknown>> = ({
 
   const [token, setToken] = useState<string | null>(initialToken);
```
Comment (tsanim) [inline src/store/auth-context.tsx:72]:
``` 
 const tokenData = retrieveStoredToken();
  const [token, setToken] = useState<string | null>(tokenData?.token);
```

File: src/store/auth-context.tsx [hunk 6]
```diff
@@ -96,6 +108,7 @@ export const AuthContextProvider: React.FC<PropsWithChildren<unknown>> = ({
     isLoggedIn: userIsLoggedIn,
     login: loginHandler,
     logout: logoutHandler,
+    userInfo: userInfo,
```
Comment (tsanim) [inline src/store/auth-context.tsx:111]: No need from `contextValue` just pass it to the provider.

File: src/pages/GameplayPage/index.tsx [hunk 7]
```diff
@@ -1,5 +1,175 @@
...
+  };
+  const getTeamPlayers = async () => {
+    if (gameId) {
+      const response = await getTeamsWithPlayers(gameId);
+      const teamPlayers: Player[][] = response.map(
+        (team) => team.players || []
+      );
+      setTeams(response);
+      const orderedPlayers: Player[] = teamPlayers
+        .map((player) => player && player[0])
```
[the remark this answers is not in the document]
Comment (tsanim) [reply src/pages/GameplayPage/index.tsx:54]: Rename `player` to `players`.

[the remark this answers is not in the document]
Comment (tsanim) [reply src/pages/GameplayPage/index.tsx:54]:
```
      const orderedPlayers = teams.reduce((players, team, index) => {
        const prevTeamPlayers = team[index - 1];
        const firstPlayers = players.splice();
        const secondPlayers = [];

        return [
          ...firstPlayers,
          team.players[0],
          ...secondPlayers,
          team.players[1],
        ];
      }, []);
      
```

**Pull request closed without merging**